pkg/controller/gkenetworkparamset/gkenetworkparamset_controller_test.go

```diff
@@ -1154,3 +1180,441 @@ func (testVals *testGKENetworkParamSetController) doesGNPFinalizerExist(ctx cont
+				}
+				return true, nil
+			}).Should(gomega.BeTrue())
+		})
+	}
+}
+
+func TestSyncDefaultPodRanges(t *testing.T) {
+	gkeNetworkParamSetName := "test-paramset"
+
+	defaultNode := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: defaultNode,
+			Labels: map[string]string{
+				utilnode.NodePoolPodRangeLabelPrefix: defaultPodRange,
+			},
+		},
+	}
+	node1 := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: node1,
+			Labels: map[string]string{
+				utilnode.NodePoolPodRangeLabelPrefix: newPodRange1,
+			},
+		},
+	}
+	tests := []struct {
+		name               string
+		defaultParamSet    *networkv1.GKENetworkParamSet
+		nonDefaultParamSet *networkv1.GKENetworkParamSet
+		nodeList           []*v1.Node
+		expectedRanges     []string
+	}{
+		{
+			name:            "No node",
+			defaultParamSet: newL3GNP(networkv1.DefaultPodNetworkName, ensureExistsMode, []string{defaultPodRange}),
+			expectedRanges:  []string{defaultPodRange},
+		},
+		{
+			name:            "No pod range label",
+			defaultParamSet: newL3GNP(networkv1.DefaultPodNetworkName, ensureExistsMode, []string{defaultPodRange}),
+			nodeList: []*v1.Node{defaultNode, {
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "no-label-node",
+					Labels: map[string]string{},
+				},
+			}},
+			expectedRanges: []string{defaultPodRange},
+		},
+		{
+			name:            "Pod range label returns empty",
+			defaultParamSet: newL3GNP(networkv1.DefaultPodNetworkName, ensureExistsMode, []string{defaultPodRange}),
+			nodeList: []*v1.Node{defaultNode, {
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "empty-label-value-node",
+					Labels: map[string]string{
+						utilnode.NodePoolPodRangeLabelPrefix: "",
+					},
+				},
+			}},
+			expectedRanges: []string{defaultPodRange},
+		},
+		{
+			name:            "No-opt in Reconcile mode",
+			defaultParamSet: newL3GNP(networkv1.DefaultPodNetworkName, reconcileMode, []string{defaultPodRange}),
+			nodeList:        []*v1.Node{node1},
+			expectedRanges:  []string{defaultPodRange},
+		},
+		{
+			name:            "Default params udpate PodIPv4Range basing on the nodes",
+			defaultParamSet: newL3GNP(networkv1.DefaultPodNetworkName, ensureExistsMode, []string{defaultPodRange, newPodRange2}),
+			nodeList:        []*v1.Node{node1},
+			expectedRanges:  []string{defaultPodRange, newPodRange1},
+		},
+		{
+			name:            "PodIPv4Range order does not matter",
+			defaultParamSet: newL3GNP(networkv1.DefaultPodNetworkName, ensureExistsMode, []string{newPodRange1, defaultPodCIDR}),
+			nodeList:        []*v1.Node{node1, defaultNode},
+			expectedRanges:  []string{defaultPodRange, newPodRange1},
+		},
+		{
+			name:               "Non-default params should not update",
+			defaultParamSet:    newL3GNP(networkv1.DefaultPodNetworkName, ensureExistsMode, []string{defaultPodRange}),
+			nonDefaultParamSet: newL3GNP(gkeNetworkParamSetName, ensureExistsMode, []string{newPodRange1}),
```

Comment:
Wouldn't this be a better test case if the range here was newPodRange2?

Reply:
Done. good point